src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -127,7 +132,10 @@ public async Task<BundleResponse> Handle(BundleRequest bundleRequest, Cancellati
             var bundleResource = bundleRequest.Bundle.ToPoco<Hl7.Fhir.Model.Bundle>();
             _bundleType = bundleResource.Type;
 
-            await FillRequestLists(bundleResource.Entry);
+            // For resources within a transaction, we need to validate if they are referring to each other and throw an exception in such case.
+            await _transactionBundleValidator.ValidateBundle(bundleResource, cancellationToken);
+
+            await FillRequestLists(bundleResource.Entry, cancellationToken);
```

Comment:
I'm now thinking it might make more sense to move both these methods into the if/else, e.g:
```csharp
if (_bundleType == BundleType.Batch)
{
    await FillRequestLists(bundleResource.Entry, cancellationToken);
    // ...
}
else if (_bundleType == BundleType.Transaction)
{
    await _transactionBundleValidator.ValidateBundle(bundleResource, cancellationToken);
    await FillRequestLists(bundleResource.Entry, cancellationToken);
    // ...
}
```